src/utils.py

```diff
@@ -486,3 +487,64 @@ def set_test_env(images, images_env="DLC_IMAGES", **kwargs):
 def get_codebuild_project_name():
     # Default value for codebuild project name is "local_test" when run outside of CodeBuild
     return os.getenv("CODEBUILD_BUILD_ID", "local_test").split(":")[0]
+
+def get_safety_ignore_dict(image_uri):
+    """
+    Get a dict of known safety check issue IDs to ignore, if specified in file ../data/ignore_ids_safety_scan.json.
+    :param image_uri:
+    :return: <list> list of safety check IDs to ignore
+    """
+    framework = ("mxnet" if "mxnet" in image_uri else
+                 "pytorch" if "pytorch" in image_uri else
+                 "tensorflow")
+    job_type = ("training" if "training" in image_uri else
+                "inference-eia" if "eia" in image_uri else
+                "inference-neuron" if "neuron" in image_uri else
+                "inference")
```

Comment:
Can this be made more future proof so that we don't have to keep adding more `if-else` conditions as more device types are added here?

Reply:
I have made it future proof to the extent possible.  According to my understanding the job_type cannot be made completely future proof because inference type images have not been further divided into inference-eia and inference-neuron in the buildspec files.